src/main/java/io/fabric8/jenkins/openshiftsync/JenkinsUtils.java

```diff
@@ -268,125 +297,143 @@ else if (envKeys.contains(param.getName())) {
+            updateSourceCredentials(buildConfig);
+
+            // We need to ensure that we do not remove
+            // existing Causes from a Run since other
+            // plugins may rely on them.
+            List<Cause> newCauses = new ArrayList<>();
+            newCauses.add(new BuildCause(build, bcProp.getUid()));
+            CauseAction originalCauseAction = BuildToActionMapper
+                    .removeCauseAction(build.getMetadata().getName());
+            if (originalCauseAction != null) {
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    LOGGER.fine("Adding existing causes...");
+                    for (Cause c : originalCauseAction.getCauses()) {
+                        LOGGER.fine("orginal cause: " + c.getShortDescription());
+                    }
+                }
+                newCauses.addAll(originalCauseAction.getCauses());
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    for (Cause c : newCauses) {
+                        LOGGER.fine("new cause: " + c.getShortDescription());
+                    }
+                }
+            }
+
+            List<Action> buildActions = new ArrayList<>();
+            CauseAction bCauseAction = new CauseAction(newCauses);
+            buildActions.add(bCauseAction);
+
+            GitBuildSource gitBuildSource = build.getSpec().getSource()
+                    .getGit();
+            SourceRevision sourceRevision = build.getSpec().getRevision();
+
+            if (gitBuildSource != null && sourceRevision != null) {
+                GitSourceRevision gitSourceRevision = sourceRevision.getGit();
+                if (gitSourceRevision != null) {
+                    try {
+                        URIish repoURL = new URIish(gitBuildSource.getUri());
+                        buildActions.add(new RevisionParameterAction(
+                                gitSourceRevision.getCommit(), repoURL));
+                    } catch (URISyntaxException e) {
+                        LOGGER.log(SEVERE, "Failed to parse git repo URL"
+                                + gitBuildSource.getUri(), e);
+                    }
+                }
+            }
+
+            ParametersAction userProvidedParams = BuildToActionMapper
+                    .removeParameterAction(build.getMetadata().getName());
+            // grab envs from actual build in case user overrode default values
+            // via `oc start-build -e`
+            JenkinsPipelineBuildStrategy strat = build.getSpec().getStrategy()
+                    .getJenkinsPipelineStrategy();
+            // only add new param defs for build envs which are not in build
+            // config envs
+            Map<String, ParameterDefinition> paramMap = addJobParamForBuildEnvs(
+                    job, strat, false);
+            // TODO do we want to abort the launching of the job if
+            // verifyEnvVars returns false?
```

Comment:
for now i'd say probably not, just logging it is good.

Reply:
agreed - thanks